internal/sender/sender.go

```diff
@@ -63,6 +110,16 @@ func (s *HTTPSender) SendMetric(m interface{}, path string) error {
 	if err != nil {
 		return err
 	}
+	if s.CryptoKey != nil {
+		encryptedPayload, err1 := Encrypt(jsonBytes, s.CryptoKey)
+		if err1 != nil {
+			return err1
+		}
+		jsonBytes, err = json.Marshal(encryptedPayload)
+		if err != nil {
+			return err
```

Comment:
Ошибки из сторонних библиотек принято оборачивать в дополнительный контекст:

```suggestion
			return fmt.Errorf("failed to marshal the encrypted payload: %w", err)
```